src/lib/process/index.ts

```diff
@@ -49,30 +51,51 @@ export const createProcess = async (
   userProgram: Process.Program,
   dryRun: boolean = false,
   options: Polkadot.Options = defaultOptions
-): Promise<Process.Result> => {
-  const program: Process.Program = validate(userProgram)
-  const processId = utf8ToHex(name, Constants.PROCESS_ID_LENGTH)
-  const polkadot: Polkadot.Polkadot = await createNodeApi(options)
-  const expectedVersion: number = (await getVersion(polkadot, processId)) + 1
+): Promise<Process.Result | Process.ProcessError> => {
+  try {
 
-  if (version !== expectedVersion) {
-    throw new VersionError(`Version: ${version} must be incremented to: ${expectedVersion}`)
-  }
+    const program: Process.Program = validate(userProgram)
+    const processId = utf8ToHex(name, Constants.PROCESS_ID_LENGTH)
+    const polkadot: Polkadot.Polkadot = await createNodeApi(options)
+    const currentVersion: number = await getVersion(polkadot, processId)
+    const expectedVersion: number = currentVersion + 1
 
-  if (dryRun)
-    return {
-      process: null,
-      message: 'Dry run: transaction has not been created',
-      name,
-      version: expectedVersion,
-      program,
+    if (version > expectedVersion || version < currentVersion)
+      throw new VersionError(version, expectedVersion, name)
+
+    if (version === currentVersion) {
+      const process = await getProcess(polkadot, processId, version)
+
+      if (program.length !== process.program.length) 
+        throw new ProgramError('existing: programs are different lengths', process) 
+
+      if (!program.every((step, i) => textify(step) === textify(process.program[i])))
```

Comment:
I tried this out by submitting `processFlows.json` from `dscp-matchmaker-api` twice in a row. It doesn't correctly validate that the process is the same. Seems to be for 2 reasons:

- the process returned from the node has some hex values. Can be fixed by using `toHuman()` rather than `toJSON` in `getProcess`
- numbers in the process from the node are wrapped in `''` whereas the user submitted program numbers are without `''`